static/js/components/ObservabilityPreferences.jsx

```diff
@@ -1,59 +1,38 @@
 import React from "react";
 import { useSelector, useDispatch } from "react-redux";
-
-import { makeStyles, useTheme } from "@material-ui/core/styles";
-import Select from "@material-ui/core/Select";
-import MenuItem from "@material-ui/core/MenuItem";
-import InputLabel from "@material-ui/core/InputLabel";
-import FormControl from "@material-ui/core/FormControl";
-import Chip from "@material-ui/core/Chip";
-import Input from "@material-ui/core/Input";
 import * as profileActions from "../ducks/profile";
 import UserPreferencesHeader from "./UserPreferencesHeader";
-
-const useStyles = makeStyles((theme) => ({
-  formControl: {
-    marginTop: theme.spacing(1),
-    minWidth: "12rem",
-  },
-  chips: {
-    display: "flex",
-    flexWrap: "wrap",
-  },
-  chip: {
-    margin: 2,
-  },
-}));
-
-const getStyles = (telescopeID, telescopeIDs = [], theme) => ({
-  fontWeight:
-    telescopeIDs.indexOf(telescopeID) === -1
-      ? theme.typography.fontWeightRegular
-      : theme.typography.fontWeightMedium,
-});
+import SelectWithChips from "./SelectWithChips";
 
 const ObservabilityPreferences = () => {
-  const classes = useStyles();
-  const theme = useTheme();
   const profile = useSelector((state) => state.profile.preferences);
   const { telescopeList } = useSelector((state) => state.telescopes);
 
   const dispatch = useDispatch();
 
+  const telescopeNametoID = { "Clear selections": "-1" };
+  const telescopeIDToName = { "-1": "Clear selections" };
+
   const handleChange = (event) => {
     const prefs = {
-      observabilityTelescopes:
-        // -1 is used for the "Clear selections" option
-        event.target.value.includes(-1) ? [] : event.target.value,
+      observabilityTelescopes: event.target.value.includes("Clear selections")
+        ? []
+        : event.target.value.map(
+            (telescopeName) => telescopeNametoID[telescopeName]
+          ),
     };
     dispatch(profileActions.updateUserPreferences(prefs));
   };
 
   telescopeList?.sort((a, b) => (a.name < b.name ? -1 : 1));
-  const telescopeIDToName = { "-1": "Clear selections" };
   telescopeList?.forEach((telescope) => {
+    telescopeNametoID[telescope.name] = telescope.id;
     telescopeIDToName[telescope.id] = telescope.name;
   });
+  const telescopeNameList = telescopeList
+    .filter((telescope) => telescope.fixed_location)
```

Comment:
Why filter only on fixed telescopes?

Reply:
It's what [was there previously.](https://github.com/skyportal/skyportal/blob/main/static/js/components/ObservabilityPreferences.jsx#L94) Do you think it should be changed?

Reply:
Wait I didn't notice this was in the context of observability. It makes sense now.